databases/week3/week3.sql

```diff
@@ -0,0 +1,135 @@
+INSERT INTO review(title,description,meal_id,stars,created_date)VALUES('Super','It was really tasty',1,5,'2021-10-01');
+-- Get all meals
+SELECT *
+from meal;
+-- Add a new meal
+INSERT INTO meal (title,description,location,`when`,max_reservations,price,created_date)VALUES('chickensandwitch','chicken strips,cream and fresh salads','bagsvard','2021-10-03 12:00:00',4,'320','2021-10-02 13:00:00');
+-- Get a meal with any id, fx 1
+SELECT *
+from meal
+WHERE id = 4;
+-- Update a meal with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE meal
+SET title = 'Vegpasta',description = 'pasta,souse and veg',price = 95
+WHERE id = 7;
+-- Delete a meal with any id, fx 1
+DELETE FROM meal
+WHERE id = 11;
+-- Get all reviews
+SELECT *
+FROM review;
+-- Add a new review
+INSERT INTO review(`title`,`description`,`meal_id`,`stars`,`created_date`)VALUES('good','It was really good',3, 5,'2021-10-05');
+-- Get a review with any id, fx 1
+SELECT *
+from review
+WHERE id = 9;
+-- Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE review
+set title = 'avg',description = 'ok',stars = 2
+WHERE id = 5;
+-- Delete a review with any id, fx 1
+DELETE FROM review
+WHERE id = 11;
+-- Get all reservations
+SELECT *
+from reservation;
+-- Add a new reservation
+INSERT INTO reservation (number_of_guests,meal_id, created_date,contact_phonenumber,contact_name,contact_email)VALUES(3,2,'2021-10-05','444444','sunil','sunil@mail.com');
+-- Get a reservation with any id, fx 1
+SELECT *
+FROM reservation
+WHERE id = 2;
+-- Update a reservation with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE reservation
+set number_of_guests = 6,meal_id = 6
+WHERE id = 5;
+-- Delete a reservation with any id, fx 1
+DELETE FROM reservation
+WHERE id = 11;
+-- Get meals that has a price smaller than a specific price fx 90
+SELECT *
+from meal
+WHERE price < 200;
+-- Get meals that still has available reservations
+SELECT *
+FROM meal
+WHERE max_reservations > (SELECT SUM(number_of_guests) FROM reservation WHERE meal_id = meal.id);
+-- Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+SELECT *
+FROM meal
+WHERE title LIKE '%Rød grød med%';
+-- Get meals that has been created between two dates
+SELECT *
+FROM meal
+WHERE created_date BETWEEN '2021-10-01'
+  AND '2021-10-02';
+-- Get only specific number of meals fx return only 5 meals
+SELECT *
+FROM meal
+LIMIT 5;
+-- Get the meals that have good reviews
+SELECT *
+FROM meal
+JOIN review ON meal.id = review.meal_id
+AND review.title = 'good';
+-- Get reservations for a specific meal sorted by created_date
+SELECT meal.id,meal.title,reservation.meal_id,reservation.created_date
+FROM meal,reservation
+WHERE meal.title LIKE "Vegburger"
+and meal.id = reservation.meal_id
+ORDER BY reservation.created_date;
```

Comment:
Why you need to test meal.title ? Can you modify it like this and see what it gives out

SELECT meal.id, reservation.meal_id, meal.title, 
reservation.number_of_guests, reservation.created_date FROM meal, reservation
WHERE  meal.id=reservation.meal_id
ORDER BY reservation.created_date;